core/src/main/java/de/jplag/reporting/reportobject/ReportObjectFactory.java

```diff
@@ -224,6 +229,24 @@ private void writeSubmissionIndexFile(JPlagResult result, String path) {
         jsonFileWriter.writeFile(fileIndex, path, SUBMISSION_FILE_INDEX_FILE_NAME);
     }
 
+    private void writeOptionsFiles(JPlagOptions options, String path) {
```

Comment:
This could lead to errors in the future, if we forget to add new options. Do you have an idea to retrieve that from the options automatically?

Reply:
Nevertheless, I like the idea of persisting this information :)

Reply:
We could add the anotations (`@JsonProp`) directly to the options and then write custom Serializers for all non native supported types (like the language) (see [https://www.baeldung.com/jackson-annotations#7-jsonserialize](https://www.baeldung.com/jackson-annotations#7-jsonserialize))

Reply:
I think that would be a good idea :)